From 481178f249e965c593db7bc8f31dd5fc3b5427d0 Mon Sep 17 00:00:00 2001
From: Benjamin Kaduk <kaduk@mit.edu>
Date: Mon, 16 Oct 2017 16:53:22 -0500
Subject: [PATCH 5/6] Correct m4 conditionals in curses.m4

AS_IF does not invoke the test(1) shell builtin for us, so we must
take care to consistently use it ourself.

While here, sprinkle some missing double-quotes around variable
expansions in AS_IF statements in this file.

Submitted by Bastian Beischer.

FIXES 134414

Change-Id: Iccfe311011f17de6317cf64abdc58b0812b81b8c
Reviewed-on: https://gerrit.openafs.org/12738
Reviewed-by: Michael Meffie <mmeffie@sinenomine.net>
Reviewed-by: Benjamin Kaduk <kaduk@mit.edu>
Tested-by: Benjamin Kaduk <kaduk@mit.edu>
(cherry picked from commit e0c5ada214596d5adb6798682d5e280cc99f447c)
Reviewed-on: https://gerrit.openafs.org/12739
(cherry picked from commit bc384b7d5e4818c567a64fe4a935f021d936444f)
---
 src/cf/curses.m4 | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/cf/curses.m4 b/src/cf/curses.m4
index a0cdc60f1..dc01469f5 100644
--- a/src/cf/curses.m4
+++ b/src/cf/curses.m4
@@ -55,7 +55,7 @@ AC_DEFUN([_OPENAFS_CURSES_GETMAXYX_BSD43],
  dnl possibly this may need to be done as above in some cases?
  AC_CHECK_FUNCS([getmaxx getmaxy], [], [_openafs_curses_bsd43=no])
  LIBS="$save_LIBS"
- AS_IF([$_openafs_curses_bsd43 = yes], [$1], [$2])])
+ AS_IF([test "$_openafs_curses_bsd43" = yes], [$1], [$2])])
 
 dnl _OPENAFS_CURSES_GETMAXYX_BSDVI([ACTION-IF-SUCCESS], [ACTION_IF_FAILURE])
 dnl test for getmaxx() and getmaxy() as from BSD curses as bodily ripped
@@ -68,7 +68,7 @@ AC_DEFUN([_OPENAFS_CURSES_GETMAXYX_BSDVI],
  AC_CHECK_MEMBERS([WINDOW._maxx WINDOW._maxy], [], [_openafs_curses_bsdvi=no],
 		  _OPENAFS_CURSES_HEADERS)
  LIBS="$save_LIBS"
- AS_IF([$_openafs_curses_bsdvi = yes], [$1], [$2])])
+ AS_IF([test "$_openafs_curses_bsdvi" = yes], [$1], [$2])])
 
 dnl OPENAFS_CURSES_WINDOW_EXTENTS([ACTION-IF-SUCCESS], [ACTION_IF_FAILURE])
 dnl check for standard getmaxyx macro. failing that, try the
@@ -78,11 +78,11 @@ AC_DEFUN([OPENAFS_CURSES_WINDOW_EXTENTS],
 [OPENAFS_CURSES_LIB
  openafs_curses_extent=none
  _OPENAFS_CURSES_GETMAXYX_XTI([openafs_curses_extent=xti])
- AS_IF([test $openafs_curses_extent = none],
+ AS_IF([test "$openafs_curses_extent" = none],
        [_OPENAFS_CURSES_GETMAXYX_BSD43([openafs_curses_extent=bsd])])
- AS_IF([test $openafs_curses_extent = none],
+ AS_IF([test "$openafs_curses_extent" = none],
        [_OPENAFS_CURSES_GETMAXYX_BSDVI([openafs_curses_extent=vi])],)
- AS_IF([test $openafs_curses_extent != none], [$1], [$2])])
+ AS_IF([test "$openafs_curses_extent" != none], [$1], [$2])])
 
 dnl The top level curses group
 AC_DEFUN([OPENAFS_CURSES],
-- 
2.15.0

